src/saf.rs

```diff
@@ -1,40 +1,157 @@
 //! A safe api into the Spatial Audio Framework.
 
-use std::ptr::{addr_of_mut, null_mut};
+use std::ptr::{addr_of_mut, null, null_mut};
 
 use crate::saf_raw;
 
 use libc::c_void;
 
+// Sets all audio channel distances to 1 meter—— stretch goal to specify per channel
+const DIST_DEFAULT: f32 = 1.0;
+const SAMP_RATE: i32 = 44100;
+const NUM_OUT_CHANNELS: usize = 2;
+const FRAME_SIZE: usize = 128;
+
+const RAD_TO_DEGREE: f32 = 180.0 / std::f32::consts::PI;
+
+///
 /// A Binauraliser is anything that can take an array of sound buffers, paired
 /// with their associated metadata, and return a pair of freshly allocated
 /// buffers representing the mixed stereo audio.
+///
 pub trait Binauraliser {
-    fn process(&self, buffers: &[(BufferMetadata, &[f32])]) -> (Vec<f32>, Vec<f32>);
+    fn new() -> Self;
+    fn process_frame(&mut self, buffers: &[(BufferMetadata, &[f32])]) -> (Vec<f32>, Vec<f32>);
+
+    ///
+    /// Takes a slice of audio data tuples for each sound source. Each tuple
+    /// contains float sound data and a BufferMetadata, which encodes the
+    /// sound source's location, range, and gain over that frame period.
+    ///
+    fn process(&mut self, buffers: &[(BufferMetadata, &[f32])]) -> (Vec<f32>, Vec<f32>) {
+        let num_samples = buffers.iter().map(|b| b.1.len()).max().unwrap_or(0);
+        let mut final_left_vec = Vec::with_capacity(num_samples);
```

Comment:
```suggestion
        let num_samples = buffers.iter().map(|(_, src)| src.len()).max().unwrap_or(0);
```
For my readability sanity :')

Reply:
I think that is my code, happy to change it!